Gitweb:     http://git.kernel.org/linus/fe5ed91ddce85a0ed0e4f92c10b099873ef62167
Commit:     fe5ed91ddce85a0ed0e4f92c10b099873ef62167
Parent:     b8b7d791a8ff01d2380089279a69afa99115fb23
Author:     Hidetoshi Seto <seto.hidetoshi@jp.fujitsu.com>
AuthorDate: Thu Dec 3 11:33:08 2009 +0900
Committer:  H. Peter Anvin <hpa@zytor.com>
CommitDate: Wed Dec 2 21:27:32 2009 -0800

    x86, mce: don't restart timer if disabled
    
    Even it is in error path unlikely taken, add_timer_on() at
    CPU_DOWN_FAILED* needs to be skipped if mce_timer is disabled.
    
    Signed-off-by: Hidetoshi Seto <seto.hidetoshi@jp.fujitsu.com>
    Cc: Andi Kleen <andi@firstfloor.org>
    Cc: Huang Ying <ying.huang@intel.com>
    Cc: Jan Beulich <jbeulich@novell.com>
    Cc: <stable@kernel.org>
    Signed-off-by: H. Peter Anvin <hpa@zytor.com>
---
 arch/x86/kernel/cpu/mcheck/mce.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

--- a/arch/x86/kernel/cpu/mcheck/mce.c
+++ b/arch/x86/kernel/cpu/mcheck/mce.c
@@ -1991,9 +1991,11 @@ mce_cpu_callback(struct notifier_block *
 		break;
 	case CPU_DOWN_FAILED:
 	case CPU_DOWN_FAILED_FROZEN:
-		t->expires = round_jiffies(jiffies +
+		if (!mce_ignore_ce && check_interval) {
+			t->expires = round_jiffies(jiffies +
 					   __get_cpu_var(mce_next_interval));
-		add_timer_on(t, cpu);
+			add_timer_on(t, cpu);
+		}
 		smp_call_function_single(cpu, mce_reenable_cpu, &action, 1);
 		break;
 	case CPU_POST_DEAD:
